contracts/Vault.vy

```diff
@@ -92,19 +92,29 @@ def deposit(token_id: uint256, price: uint256, min_duration: uint256, max_durati
     # transfer token to this contract
     IERC721(self.nft_contract_addr).safeTransferFrom(self.owner, self, token_id, b"")
 
+    # create delegation
+    self._delegate_to_owner()
+
 
 @external
 def set_listing(sender: address, price: uint256, min_duration: uint256, max_duration: uint256):
     assert self.is_initialised, "not initialised"
     assert msg.sender == self.caller, "not caller"
     assert sender == self.owner, "not owner of vault"
```

Comment:
Should these go to the `_set_listing` method, since they're the same asserts as in the `set_listing_and_delegate_to_owner`?

Reply:
Considered that, but seems more hidden since the other checks are done in the external functions. Also for `set_listing_and_delegate_to_owner` the checks would be split. @DioPires wdyt?

Reply:
Yeah, I was also on the fence regarding this. We can leave it like it is 👍